### ISSUE TITLE
:bomb: `Bug`: dtm init can't work

### ISSUE THREAD

**Issue by huang12zheng** [devstream-io/devstream#673]
### What Happened?

whne I use `dtm init -f quickstart.yaml`,cause failure
```
2022-06-14 12:54:25 ✖ [ERROR]  Config file validation failed: tool file is empty.
2022-06-14 12:54:25 ℹ [INFO]  Using dir <.devstream> to store plugins.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4aeb477]

goroutine 1 [running]:
github.com/devstream-io/devstream/internal/pkg/pluginmanager.DownloadPlugins(0x0)
	github.com/devstream-io/devstream/internal/pkg/pluginmanager/manager.go:30 +0x177
main.initCMDFunc(0x51ffe00, {0xc000155ca0, 0x0, 0x2})
	github.com/devstream-io/devstream/cmd/devstream/init.go:26 +0x185
github.com/spf13/cobra.(*Command).execute(0x51ffe00, {0xc000155c80, 0x2, 0x2})
	github.com/spf13/cobra@v1.4.0/command.go:860 +0xb67
github.com/spf13/cobra.(*Command).ExecuteC(0x5200580)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x8b3
github.com/spf13/cobra.(*Command).Execute(0x5200580)
	github.com/spf13/cobra@v1.4.0/command.go:902 +0x2f
main.main()
	github.com/devstream-io/devstream/cmd/devstream/main.go:98 +0x32
```

### How to Reproduce?
https://asciinema.org/a/wiIVgSSFCCEW2W6bwAiopBsGk
---
# core config
varFile: "" # If not empty, use the specified external variables config file
toolFile: "" # If not empty, use the specified external tools config file
state: # state config, backend can be local or s3
  backend: local
  options:
    stateFile: devstream.state

---
# plugins config
tools:
  - name: github-repo-scaffolding-golang
    instanceID: default
    options:
      owner: huang12zheng
      repo: go-webapp-devstream-demo
      branch: main
      image_repo: hzgood/go-webapp-devstream-demo
  - name: githubactions-golang
    instanceID: default
    dependsOn: ["github-repo-scaffolding-golang.default"]
    options:
      owner: huang12zheng
      repo: go-webapp-devstream-demo
      language:
        name: go
        version: "1.17"
      branch: main
      build:
        enable: True
      test:
        enable: True
        coverage:
          enable: True
      docker:
        enable: False

### Anything else

maybe like #581 with "Config file validation failed: tool file is empty."
it happen also with 0.6.1

### DevStream Version

v0.6.0

**Comment by iyear** [comment]
The doc for v0.6.0 is correct for its `quickstart.yaml` config, and you may be viewing the latest doc.

Visit the doc for v0.6.0: https://docs.devstream.io/en/release-0.6/quickstart_en/

**Comment by huang12zheng** [comment]
> v0.6.0 的文档对其`quickstart.yaml`配置是正确的，您可能正在查看最新的文档。
> 
> [访问 v0.6.0 的文档： https](https://docs.devstream.io/en/release-0.6/quickstart_en/) ://docs.devstream.io/en/release-0.6/quickstart_en/[](https://docs.devstream.io/en/release-0.6/quickstart_en/)

it is correct. 
it is work with doc release-0.6.

**Comment by iyear** [comment]
Is it necessary to add a v0.6.1 version of the docs? According to the v0.6.1 release, the doc has been updated.

@daniel-hutao @IronCore864

**Comment by daniel-hutao** [comment]
> Is it necessary to add a v0.6.1 version of the docs? According to the v0.6.1 release, the doc has been updated.
> 
> @daniel-hutao @IronCore864

This is a problem, I'll put up the 0.6.0 documentation right away.

**Comment by iyear** [comment]
@daniel-hutao Some errors have occurred! The link above is no longer accessible. The doc for v0.6.0 is now: https://docs.devstream.io/en/v0.6.0/quickstart_en/

But its quickstart is wrong. Its link points to the main branch.

**Comment by huang12zheng** [comment]
> I don't know if I got it right
> The issue's wrong is:
doc's latest Version is points to main branch, but main is not released.
So, actually, the lastest of the document should be pointing to the latest release.
@iyear @daniel-hutao

**Comment by iyear** [comment]
@huang12zheng In the current design, the latest page of the docs points to the latest main branch code instead of the latest release.

I think it is possible to have a default page for `docs.devstream.io` that points to the latest release instead of the docs for the latest main branch code.

**Comment by huang12zheng** [comment]
> @huang12zheng In the current design, the latest page of the docs points to the latest main branch code instead of the latest release.
> 
> I think it is possible to have a default page for `docs.devstream.io` that points to the latest release instead of the docs for the latest main branch code.

Pretty much what I said, although I didn't find solution that add a default page

**Comment by daniel-hutao** [comment]
@IronCore864 How about we switch the default document version to the latest release version？

**Comment by huang12zheng** [comment]
@daniel-hutao 
I just want to say that the documentation version is not as simple.

* Maybe we should learn from how other projects manage documentation

Some demon whisper, I'm just in confusion:
There are the following possible facts:
~~1. In the future, a document error is a low probability event,
   Even documents don't change very much~~
~~2. We may need to decouple documents from releases,
3. Or we don't need to maintain past documents
Just maintain the documents that stable mainly supports and the latest versions of the documents.~~
1. Update documents and release new versions. We shouldn't like that
Release: 0.6.1 - > 0.6.2 - devK
Doc: 0.6.2 - devK

2. At development time, releases are locked, but documents are branch dependent.
Doc - 0.6.1 ~ release 0.6.1
We only maintain the latest stable version of the document.

* If you have a good plan, you might ignore what I'm saying

**Comment by IronCore864** [comment]
@huang12zheng we did some changes regarding documentation and releases. To sum up:

- We follow [this doc](https://docs.devstream.io/en/latest/development/branch-and-release/) for releases and branches. TL;DR: we will try our best to support the latest 3 versions. Given our current shortage of manpower, this can't be guaranteed at the moment, but we will strive for it.
- Documentation versions: now that the v0.7 is released, we made the v0.7 version the default version when you open docs.devstream.io. The "latest" version won't be shown as default anymore.

If you are interested, you can give the quickstart a try according to the v0.7 documentation. Let us know if there are any more issues. Thanks very much for your contribution to this matter. Closing this issue for now, but do re-open or create a new one if you find more issues according to the v0.7 doc.
